### PR TITLE
Extract Process Variables

### PR DESCRIPTION
**Pull request by pinussilvestrus**
This offers two API methods
* `getProcessVariables` - extracts _all_ variables which can be found inside a root element (e.g. a `bpmn:Process`)
* `getVariablesForScope`  - retrieves all variables _available_ inside a given scope (also parent scope variables)

Note that this is independent from `diagram-js`, so it will only work with moddle elements/business objects.
Note that some utils from other projects were used and modified for the purpose of this library (e.g. the `InputOutputHelper` from `bpmn-js-properties-panel`). The goal is to not have dependencies to other projects, that's why they were copied and re-written.

__Which issue does this PR address?__

Related to bpmn-io/bpmn-js-properties-panel#342

__Acceptance Criteria__

* [ ] Corresponds to the concept (bpmn-io/bpmn-js-properties-panel#342)
* [x] <del>Corresponds to the design</del>

__Definition of Done__

* [ ] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [ ] corresponds to [the code standards](https://github.com/bpmn-io/.github/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [ ] passes Continuous Integration checks
* [ ] available as feature branch on GitHub
  * [ ] contains the cleaned up commit history
  * [ ] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [ ] a single commit closes the issue via Closes #issuenr

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
Task(s) to be done
* [ ] Add Usage instructions

File: package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "@bpmn-io/extract-process-variables",
+  "version": "0.0.0",
```
Comment (barmac) [inline package.json:3]:
```suggestion
  "version": "0.0.0",
```
Let's update this before release 😅

File: package.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  "author": "Niklas Kiefer <niklas.kiefer@camunda.com> (https://github.com/pinussilvestrus)",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/bpmn-io/extract-process-variables/issues"
+  },
+  "homepage": "https://github.com/bpmn-io/extract-process-variables#readme",
+  "devDependencies": {
+    "@rollup/plugin-json": "^4.1.0",
+    "bpmn-moddle": "^7.0.3",
+    "camunda-bpmn-moddle": "^4.4.0",
```
Comment (barmac) [inline package.json:31]: This is also a peer dependency, right?

File: test/spec/ProcessVariablesHelperSpec.js [hunk 3]
```diff
@@ -0,0 +1,321 @@
+import chai, { expect } from 'chai';
+
+import chaiAsPromised from 'chai-as-promised';
+
+chai.use(chaiAsPromised);
```
Comment (barmac) [inline test/spec/ProcessVariablesHelperSpec.js:5]: Do we use that anywhere?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+language: node_js
+node_js: 'node'
+
+script:
+  - npm run all
```
Comment (barmac) [inline .travis.yml:5]: We're going to need this for test coverage reports:

```suggestion
  - COVERAGE=1 npm run all
  - npx codecov
```

File: package.json (hunk 1 above)
Comment (pinussilvestrus) [reply package.json:3]: wouldn't it be updated by np automatically?

File: package.json (hunk 2 above)
Comment (pinussilvestrus) [reply package.json:31]: Jep, thanks 👍

File: test/spec/ProcessVariablesHelperSpec.js (hunk 3 above)
Comment (pinussilvestrus) [reply test/spec/ProcessVariablesHelperSpec.js]: Nope, I removed it 👍

File: package.json (hunk 1 above)
Comment (barmac) [reply package.json:3]: Could be. We can leave it as it is.

**Pull request merged**